addons/adyen_platforms/models/adyen_account.py

```diff
@@ -213,75 +284,93 @@ def action_create_redirect(self):
-                    'dateOfBirth': str(self.date_of_birth),
+
+        if 'email' in values:
+            holder_details['email'] = values['email']
+
+        if 'phone_number' in values:
+            holder_details['fullPhoneNumber'] = values['phone_number']
+
+        if 'is_business' in values:
+            data['legalEntity'] = 'Business' if values['is_business'] else 'Individual'
+
+        if (values.get('is_business') or self.is_business) and {'legal_business_name', 'doing_business_as', 'registration_number'} & fields:
+            business_details = holder_details['business_details'] = {}
+            for source, dest in [
+                ('legal_business_name', 'legalBusinessName'),
+                ('doing_business_as', 'doingBusinessAs'),
+                ('registration_number', 'registrationNumber'),
+            ]:
+                if source in values:
+                    business_details[dest] = values[source]
+
+        elif {'first_name', 'last_name', 'date_of_birth', 'document_number', 'document_type'} & fields:
+            holder_details['individualDetails'] = {}
+
+            if {'first_name', 'last_name'} & fields:
+                holder_details['individualDetails']['name'] = {
+                    'firstName': values.get('first_name') or self.first_name,
+                    'lastName': values.get('last_name') or self.last_name
                 }
-            }
 
-            # documentData cannot be present in the data if not set
-            if self.document_number:
-                data['accountHolderDetails']['individualDetails']['personalData']['documentData'] = [{
-                    'number': self.document_number,
-                    'type': self.document_type,
+            if 'date_of_birth' in fields:
+                holder_details['individualDetails'].setdefault('personalData', {})['dateOfBirth'] = str(values['date_of_birth'])
+
+            document_number = values.get('document_number') or self.document_number
+            if self.document_number and 'document_number' in fields:
+                holder_details['individualDetails'].setdefault('personalData', {})['documentData'] = [{
+                    'number': document_number,
+                    'type': values.get('document_type') or self.document_type,
                 }]
 
+        if holder_details:
+            data['accountHolderDetails'] = holder_details
+
         return data
 
-    def _adyen_rpc(self, operation, adyen_data={}):
-        if operation == 'create_account_holder':
+    def _adyen_rpc(self, operation, adyen_data=None):
+        adyen_data = adyen_data or {}
+        if operation == 'v1/create_account_holder':
             url = self.env['ir.config_parameter'].sudo().get_param('adyen_platforms.onboarding_url')
             params = {
                 'creation_token': request.session.get('adyen_creation_token'),
+                'base_url': self.env['ir.config_parameter'].sudo().get_param('web.base.url'),
```

Comment:
As https://github.com/odoo/odoo/pull/68201 was actually merged, and you just push forces some changes, could you adapt it to use `get_base_url()` if you need another push force? It now support empty recordset
Otherwise, you can create if afterward if too annoying to do it here

Reply:
Thanks for #68201!
I'll make the change tomorrow to not delay any longer this PR :-D